src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -495,8 +496,7 @@ static bool simplify_log(shared_ptr<Node> n)
 // other cases into Concat of shapeof/gather(data) + shapeof(indices)
 static bool simplify_gather_shapeof(shared_ptr<Node> node)
 {
-    auto shapeof = as_type_ptr<opset3::ShapeOf>(node);
-    auto gather = as_type_ptr<opset3::Gather>(shapeof->input_value(0).get_node_shared_ptr());
+    auto gather = as_type_ptr<opset3::Gather>(node->input_value(0).get_node_shared_ptr());
```

Comment:
Is this Optimization intended to apply only for `gather::v3` ? , what if we need apply the same optimization for `gather::v2`? with `ShapeOf::v2/v3` ?

Reply:
these resolve to underlying op version; This pass does not do any opset upgrade/downgrade, just handles given op.

All new code will try to use latest opset, at the same time support optimizing older opsets (as much as possible). As part of rewriting/creating subgraph we use new opset, even if we are replacing older opset.

Reply:
okay, but if their is new Opset and If we need apply to the same Optimization, we will end up reimplementing isn't it?, since this one will only work for the specific Opset. I understand for the current requirement this works.

Reply:
no, new opset does not mean op version will change. so we may not need to redo this.

Reply:
And if the op does change, the optimization may need to be changed as well since the current optimization may be incorrect for the new semantics.